### DISCUSSION TITLE
newlib C aligned_alloc implementation issue

### DISCUSSION THREAD

**Discussion by davidsj2** [zephyrproject-rtos/zephyr#62263] [Q&A]
The implementation `aligned_alloc` for newlib C library appears to make use of `posix_memalign` for some targets (tested riscv and leon3 qemu targets), which doesn't exist.  This leads to undefined references:

```
aligned_alloc.c:(.text.aligned_alloc+0xa): undefined reference to `posix_memalign'
collect2: error: ld returned 1 exit status
```

Currently using SDK version 0.16.1 and latest Zephyr master revision (0df085bb36f46dd7f9f378de7b84ccf4ee91ce28)

**Comment by m-torhan** [comment]
Did you define `CONFIG_HEAP_MEM_POOL_SIZE` in your project config?
